### PR TITLE
Improve `RegexBasedFilter` doc

### LINKED ISSUE #1269: There is no mention of how to use RegexBasedFilter to match desired branches and ignore branches with pipeline script
### Describe your use-case which is not covered by existing documentation.

There is no mention of how to use RegexBasedFilter to match desired branches and ignore branches that do not need to trigger a build. There is no mention of using pipeline scripte to configure RegexBasedFilter. I found some useful configurations in StackOverflow but it doesn't come from official documentation, such as:
https://stackoverflow.com/questions/30095583/how-do-i-configure-jenkins-to-build-all-branches-except-a-few-which-i-exclude

Useful scripts are as follows:
```groovy
triggers {
    gitlab(
        triggerOnPush: true, 
        triggerOnMergeRequest: false, 
        branchFilterType: "RegexBasedFilter", 
        targetBranchRegex: '^(?:(?!feature).)*$'
    )
}
```
You can see that the `targetBranchRegex` variable is not mentioned in the documentation at all.

Can someone fix this missing piece? Thank you for all.

### Reference any relevant documentation, other materials or issues/pull requests that can be used for inspiration.

_No response_

### PR DESCRIPTION
**Pull request by harsh-ps-2003**
Adding missing examples and some explanation related to regular expressions used in `RegexBasedFilter`.

Closes #1269.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -482,7 +482,30 @@ This plugin can be used with Matrix/Multi-configuration jobs together with the [
...
+    (: Starts a group that will be used for the negative lookahead.
+    ?!: Indicates a negative lookahead assertion - find all that doesnt match.
+    .*: Matches any number of characters (except for newline) zero or more times.
+    master: should not match master.
+    ): Ends the group.
+    $: Anchors the match to the end of the string.
+    
+Keep in mind that the RegexBasedFilter feature is case-sensitive by default. If you want to make it case-insensitive, you can use the `(?i)` flag at the beginning of your regular expression pattern. For example: `^(?i)(?:(?!feature).)*$`.
+    
+Here is an example pipeline script that shows how to use the RegexBasedFilter feature in the gitlab trigger:
```
Comment (krisstern) [inline README.md]:
```suggestion
Here is an example pipeline script that shows how to use the RegexBasedFilter feature in the GitLab trigger:
```

File: README.md [hunk 2]
```diff
@@ -482,7 +482,30 @@ This plugin can be used with Matrix/Multi-configuration jobs together with the [
...
+Triggers may be filtered based on the branch name, i.e. the build will only be allowed for selected branches. On the project configuration page, when you configure the GitLab trigger, you can choose 'Filter branches by name' or 'Filter branches by regex.' Filter by name takes comma-separated lists of branch names to include and/or exclude from triggering a build. Filter by regex takes a Java regular expression to include and/or exclude. For example, to exclude all branches containing the word "feature", you can use the following regular expression: `^(?:(?!feature).)*$`.
+On a similar note, the regular expression `^(?!.*master).*$` will mean - all branches not matching master. This is a regular expression that uses negative lookahead to match any string that does not contain the word "master". Here's a breakdown of how it works:
+
+    ^: Anchors the match to the beginning of the string.
+    (: Starts a group that will be used for the negative lookahead.
+    ?!: Indicates a negative lookahead assertion - find all that doesnt match.
```
Comment (krisstern) [inline README.md]:
```suggestion
    ?!: Indicates a negative lookahead assertion - find all that does not match.
```

**Review by krisstern — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -482,7 +482,30 @@ This plugin can be used with Matrix/Multi-configuration jobs together with the [
...
+    .*: Matches any number of characters (except for newline) zero or more times.
+    master: should not match master.
+    ): Ends the group.
+    $: Anchors the match to the end of the string.
+    
+Keep in mind that the RegexBasedFilter feature is case-sensitive by default. If you want to make it case-insensitive, you can use the `(?i)` flag at the beginning of your regular expression pattern. For example: `^(?i)(?:(?!feature).)*$`.
+    
+Here is an example pipeline script that shows how to use the RegexBasedFilter feature in the GitLab trigger:
+
+```triggers {
```
Comment (krisstern) [inline README.md]:
````suggestion
```
triggers {
````

Comment (krisstern) [reply README.md]: Formatting is a bit off here

**Review by krisstern — CHANGES_REQUESTED** [review]

**Comment by krisstern** [comment]
Ready for a review?

**Comment by harsh-ps-2003** [comment]
> Ready for a review?

I'm sorry, did I mess up something? 😨 Anything more that you would like to see in the doc?

**Comment by krisstern** [comment]
No, but you are suppoed to tell us when we should follow up with another review, say via a message or request to review button above on the top right panel.

**Comment by harsh-ps-2003** [comment]
Oh! I thought that I would be disturbing you guys. Yup! If it's good enough. Ready for review.

File: README.md [hunk 4]
```diff
@@ -482,7 +482,31 @@ This plugin can be used with Matrix/Multi-configuration jobs together with the [
...
+
+    ^: Anchors the match to the beginning of the string.
+    (: Starts a group that will be used for the negative lookahead.
+    ?!: Indicates a negative lookahead assertion - find all that does not match.
+    .*: Matches any number of characters (except for newline) zero or more times.
+    master: should not match master.
+    ): Ends the group.
+    $: Anchors the match to the end of the string.
+    
+Keep in mind that the RegexBasedFilter feature is case-sensitive by default. If you want to make it case-insensitive, you can use the `(?i)` flag at the beginning of your regular expression pattern. For example: `^(?i)(?:(?!feature).)*$`.
```
Comment (krisstern) [inline README.md]:
```suggestion
Keep in mind that the `RegexBasedFilter` feature is case-sensitive by default. If you want to make it case-insensitive, you can use the `(?i)` flag at the beginning of your regular expression pattern. For example: `^(?i)(?:(?!feature).)*$`.
```

File: README.md [hunk 5]
```diff
@@ -482,7 +482,31 @@ This plugin can be used with Matrix/Multi-configuration jobs together with the [
...
+    
+Here is an example pipeline script that shows how to use the RegexBasedFilter feature in the GitLab trigger:
```
Comment (krisstern) [inline README.md]:
```suggestion
Here is an example pipeline script that shows how to use the `RegexBasedFilter` feature in the GitLab trigger:
```

**Review by krisstern — CHANGES_REQUESTED** [review]

File: README.md [hunk 6]
```diff
@@ -482,7 +482,31 @@ This plugin can be used with Matrix/Multi-configuration jobs together with the [
...
+Triggers may be filtered based on the branch name, i.e. the build will only be allowed for selected branches. On the project configuration page, when you configure the GitLab trigger, you can choose 'Filter branches by name' or 'Filter branches by regex.' Filter by name takes comma-separated lists of branch names to include and/or exclude from triggering a build. Filter by regex takes a Java regular expression to include and/or exclude. For example, to exclude all branches containing the word "feature", you can use the following regular expression: `^(?:(?!feature).)*$`.
+On a similar note, the regular expression `^(?!.*master).*$` will mean - all branches not matching master. This is a regular expression that uses negative lookahead to match any string that does not contain the word "master". Here's a breakdown of how it works:
+
+    ^: Anchors the match to the beginning of the string.
+    (: Starts a group that will be used for the negative lookahead.
+    ?!: Indicates a negative lookahead assertion - find all that does not match.
```
Comment (krisstern) [inline README.md]:
```suggestion
    ?!: Indicates a negative lookahead assertion - finds all that does not match.
```

File: README.md [hunk 7]
```diff
@@ -482,7 +482,31 @@ This plugin can be used with Matrix/Multi-configuration jobs together with the [
...
+    .*: Matches any number of characters (except for newline) zero or more times.
```
Comment (krisstern) [inline README.md]:
```suggestion
    .*: Matches any number of characters (except for a newline) zero or more times.
```

**Review by krisstern — CHANGES_REQUESTED** [review]

**Review by krisstern — APPROVED** [review]

**Comment by krisstern** [comment]
Thanks @harsh-ps-2003

**Pull request merged**